vllm/engine/llm_engine.py

```diff
@@ -1294,6 +1378,21 @@ def step(self) -> List[Union[RequestOutput, EmbeddingRequestOutput]]:
 
         return ctx.request_outputs
 
+    def _current_step(
+            self, seq_group_metadata_list: List[SequenceGroupMetadata]
+    ) -> Optional[int]:
+        if not seq_group_metadata_list:
+            return None
+
+        current_step = seq_group_metadata_list[0].state.current_step
+        if any([
+                seq_group.state.current_step != current_step
+                for seq_group in seq_group_metadata_list[1:]
+        ]):
+            raise AssertionError(("All running sequence groups should "
```

Comment:
We shouldn't duplicate this check everywhere as `_has_remaining_steps` already checks this

Reply:
Removed this function as this is no longer needed with some changes to the `SchedulerContext` data structure.